### PR TITLE
New plants!

### PR DESCRIPTION
**Pull request by spessbro**
## About The Pull Request

Alright, well i've added a whole lot of mutations to plants that did not have any:
-plasma garlic(new mutation for garlic that if juiced makes plasma)
-awful garlic(garlic but it has spewium)
-brig-caps (mutation for liberty caps(cut up to make handcuffs)(ferment for beepsky smash)
-truffles(mutation for chanterelles contains very good tasting reagent and can be juiced for super duper reagent)
-pizzapple(mutation for pineapple that has hawaiian pizza for a product)(the fact that it's impossible to get seeds from it is intended)
-crabbage(cabbage that contains itching powder and can make crab)(waiting for that crab rave in bar)
-lye-mes(mutation for limes that contain lye and can be used as soap)
-mushroompods(mutation for replica pods that makse mushroom people instead of podpeople)
-treant shells(mutation for replica pod that makes wooden golems instead of podpeople)

## Why It's Good For The Game

Having played a bunch of botany i've noticed you can't experiment as much with mutations and basically just have to go on the wiki instead of just trowing some mutagen on plants and seeing what comes up.

## Changelog
:cl:
add: see about paragraph
del: Removed cabbage mutation into replica pod (seemed redundant and didn't make much sense)
/:cl:

### CODE REVIEW

**Comment by spessbro** [comment]
everything should be there might wanna ping @Big Dumb Dumb Idiot on discord so i can respond quickly

**Comment by 4dplanner** [comment]
>awful garlic(garlic but it has rotatium and spewium)

>:(

**Comment by MrPerson** [comment]
>carbbage

And not crabbage?

**Comment by spessbro** [comment]
i don't know how to spell ok stop bulli

**Comment by Shapsy** [comment]
Why would there be plasma in garlic

**Comment by spessbro** [comment]
cause plasma stinks

**Comment by Ty-the-Smonk** [comment]
Mint leaves plant when

**Comment by Shadowflame909** [comment]
Why'd you delete gaseous composition

**Comment by spessbro** [comment]
because of plasma garlic being wayyyyy too annoying with it

**Comment by Nabski89** [comment]
I'm pretty sure the changelog tries to automatically put that into the change log, so in game it'll just say "see above paragraph", which is useless.

Cabbages into pod person allows you to get easier cloning on charlie station, but otherwise not that big of a loss.

You're taking mushroom people from admin only to obtainable by the crew, which is iffy, but wood golems is a cool idea.

Maybe removing something else so you can grow plasma isn't the best idea.

I think that when you copy and paste an entire section without changing anything the preferred method is to make it into a proc and call that?

**Comment by necromanceranne** [comment]
>removes gaseous decomposition
>doesn't state this in the changelog

Very cool. Not like spewium isn't a traitor chem and is pretty debilitating a chem. Hypodermic prickles with slippery skin would make it utterly horrid. Getting enough to make a smoke grenade or foam grenades, or throw into a smoke machine, would be absolutely goddamn disastrous. It's balanced around most traitors only having the 30 units they get out of the kit. Why on earth would you let botanists get it with any amount of reliability. Even plasma garlic would be absolutely devastating, so if I got one or the other I've been handed two weapons of mass murder.

You don't need gaseous decomp to even remotely put that spewium to use.

**Comment by datorangebottle** [comment]
>del:removed gaseous decomposition trait

WHY ARE WE GUTTING BOTANY AGAIN?

**Comment by spessbro** [comment]
>Even plasma garlic would be absolutely devastating
 please tell me how because i honestly don't know

**Comment by ATH1909** [comment]
> Cabbages into pod person allows you to get easier cloning on charlie station, but otherwise not that big of a loss.
> 
> You're taking mushroom people from admin only to obtainable by the crew, which is iffy, but wood golems is a cool idea.

It's a bit sad that we're losing the cabbage->replica pod mutation, but I do think that it'll save a lot of frustration in the long run for people who are trying to roll for crabbages. Maybe we could make it a mutation for another plant? The only three plants that we have right now that have pods are vanilla pods, cocoa pods, and kudzu, which don't really fit the source material outside of the pun (kudzu sort of fits?).

Mushroompeople are fine to have available to the crew; they're actually quite weak. They take extraburn damage and move more slowly, but get better-than-normal punching damage and a martial art. It should be noted that said martial art is kind of bad (especially since it keeps mushpeople from combining their decent punching damage with the boxing martial art), having only one move that can only be executed after standing still next to someone for 2.5 seconds (does the mood timer speed bonus affect that?) that just deals 15-30 damage, yeets them across the room, and paralyzes them for 2 seconds (not long enough to get off another mushpunch, especially since you just flung them across the room).

Why would you remove gaseous decomposition? Just... why? The PR was going so well, man, so well. I was so psyched that you included my "mutation of replica pods that makes wooden golems" idea, but my hopes were crushed when I saw that line of the changelog and knew that this would never get merged.

necromanceranne, it should be noted that hypodermic prickles plants can only inject up to 20u of reagents per hit/slip. You need 30u to OD and the OD effect takes a looong time to actually start to make you spew up organs (maybe this will make people use stasis beds so that doctors can reimplant their organs after they throw them up?). It's also easily counterable by just chugging some charcoal, IIRC.

**Comment by ATH1909** [comment]
> because of plasma garlic being wayyyyy too annoying with it

Is there a hacky way of making plasma production incompatible with gaseous decomposition? Maybe you could make it be stable_plasma (or whatever that reagent's called) instead?

What if, instead of making it have a plasma chem production trait, you make plasma garlic able to be sliced up into sheets of plasma (with a small burst of plasma gas being released in the process)?

**Comment by necromanceranne** [comment]
More importantly I don't think gaseous decomp even stops the plant from dumping the plasma into the air if you simply use liquid contents. So that's two genes you'd need to remove. Which also means tomatoes would make next to no sense, and you'd need to get rid of bluespace tomatoes as well.

**Comment by spessbro** [comment]
>It should be noted that said martial art is kind of bad

they don't even get boxing actually

File: code/modules/hydroponics/grown/citrus.dm [hunk 1]
```diff
@@ -32,7 +32,69 @@
...
+/obj/item/reagent_containers/food/snacks/grown/citrus/lyeme
+	seed = /obj/item/seeds/lime/lyeme
+	name = "lye-me"
+	desc = "These could defeat the most robust of stains."
+	icon_state = "lyeme"
+	filling_color = "#00FF00"
+	var/cleanspeed = 50
+/obj/item/reagent_containers/food/snacks/grown/citrus/lyeme/add_juice()
+	..()
+	cleanspeed =round(50-(seed.potency/2))
```
Comment (ATH1909) [inline code/modules/hydroponics/grown/citrus.dm]: change this to be
cleanspeed =round(**60**-(seed.potency/2))

We don't want lye-mes with a cleanspeed of 0, right?

File: code/modules/hydroponics/grown/mushrooms.dm [hunk 2]
```diff
@@ -109,6 +110,37 @@
...
+	name = "brig-cap"
+	desc = "<B>Like it or not you're comming with me!</B>"
+	icon_state = "brigcap"
+	filling_color = "#DAA520"
+	wine_power = 80
+	distill_reagent = /datum/reagent/consumable/ethanol/beepsky_smash
+/obj/item/reagent_containers/food/snacks/grown/mushroom/brigcap/attackby(obj/item/W, mob/user, params)
+	if(W.sharpness)
+		user.show_message("<span class='notice'>You make hancuffs out of \the [src]!</span>", 1)
+		new /obj/item/restraints/handcuffs(user.loc)
```
Comment (ATH1909) [inline code/modules/hydroponics/grown/mushrooms.dm]: could you make the difficulty of escaping the handcuffs scale with potency?

**Comment by necromanceranne** [comment]
So liquid plasma when spilled into a space creates an equal amount of moles of plasma gas. This also carries heat, which means you can get plasma heated up to it's flashpoint to create a plasma fire when it ignites with the oxygen in the air upon spilling. You just need some way to heat the plasma, and there are a few ways to do that, namely pyro grenades but also just clf3 dumped into a square can do the trick. By taking the gene, and putting it into a fast growing plant, you can mass produce liquid plasma without having to even touch the plasma sheet reserves at all.

**Comment by ATH1909** [comment]
> I think that when you copy and paste an entire section without changing anything the preferred method is to make it into a proc and call that?

Also, this (although I can get why you might want to play it safe and just copy+paste, it makes the code look really ugly/makes it harder to change replica pods and their kin in the future).

**Comment by ATH1909** [comment]
> > It should be noted that said martial art is kind of bad
> 
> they don't even get boxing actually

Exactly; they get their special mushroom martial art, which keeps them from learning the boxing martial art when they put on boxing gloves .

File: code/modules/hydroponics/grown/mushrooms.dm (hunk 2 above)
Comment (spessbro) [reply code/modules/hydroponics/grown/mushrooms.dm]: i can try but i might need to make brand new handcuffs

**Comment by ATH1909** [comment]
> So liquid plasma when spilled into a space creates an equal amount of moles of plasma gas. This also carries heat, which means you can get plasma heated up to it's flashpoint to create a plasma fire when it ignites with the oxygen in the air upon spilling. You just need some way to heat the plasma, and there are a few ways to do that, namely pyro grenades but also just clf3 dumped into a square can do the trick. By taking the gene, and putting it into a fast growing plant, you can mass produce liquid plasma without having to even touch the plasma sheet reserves at all.

So yeah, just make the garlic sliceable into sheets of plasma (and small bursts of plasma gas on each slice for flavor); don't give it the plasma chem production trait. Then you can bring back gaseous decomposition and everyone will be happy (right?).

**Comment by spessbro** [comment]
> > I think that when you copy and paste an entire section without changing anything the preferred method is to make it into a proc and call that?
> 
> Also, this (although I can get why you might want to play it safe and just copy+paste, it makes the code look really ugly/makes it harder to change replica pods and their kin in the future).

true it's just that i'm not very accustomed to dm code so i'm trying to really play it safe

**Comment by spessbro** [comment]
>So yeah, just make the garlic sliceable into sheets of plasma (and small bursts of plasma gas on each slice for flavor); don't give it the plasma chem production trait. Then you can bring back gaseous decomposition and everyone will be happy (right?).

what if i made it so you need to juice them to get the plasma?

File: code/modules/hydroponics/grown/citrus.dm (hunk 1 above)
Comment (spessbro) [reply code/modules/hydroponics/grown/citrus.dm]: good point

**Comment by Ty-the-Smonk** [comment]
Can wood golems be made with xenobio shells as well?

**Comment by spessbro** [comment]
> 
> 
> Can wood golems be made with xenobio shells as well?

yes

**Comment by Ty-the-Smonk** [comment]
> > Can wood golems be made with xenobio shells as well?
> 
> yes

More slime for the slime god

**Comment by knacker48** [comment]
> > > Can wood golems be made with xenobio shells as well?
> > 
> > 
> > yes
> 
> More slime for the slime god

I'm pretty sure you could always make wood golems, just that most people stick to iron since wood isn't always available

**Comment by ATH1909** [comment]
> > So yeah, just make the garlic sliceable into sheets of plasma (and small bursts of plasma gas on each slice for flavor); don't give it the plasma chem production trait. Then you can bring back gaseous decomposition and everyone will be happy (right?).
> 
> what if i made it so you need to juice them to get the plasma?

Oh yeah, I forgot that juice_results was a thing. Yeah, just add plasma to the juice_results list and keep gaseous decomposition in the game. That's a much better solution.

For those of you who don't know, putting something on the juice_results of a plant makes it appear only when a plant is juiced, in an amount proportional to the amount of nutriment in the plant (this removes the nutriment and vitamins from the juiced product).

**Comment by AdamElTablawy** [comment]
Gaseous decomp removal would be necessary if he's adding the plasma garlic, otherwise plasma smoke grenade plants would be a thing.

Spewium is a traitor chem that can make you eject your heart, so i'd be careful with that

File: code/modules/hydroponics/grown/mushrooms.dm (hunk 2 above)
Comment (spessbro) [reply code/modules/hydroponics/grown/mushrooms.dm]: yeah the new handcuffs are way too much of a hastle plus i don't want to add more codersprites than i've already done

**Comment by ExcessiveUseOfCobblestone** [comment]
I hate skewium so much man

**Comment by spessbro** [comment]
> 
> 
> I hate skewium so much man

good thing i didn't add it

**Comment by ExcessiveUseOfCobblestone** [comment]
and rotatium lol

**Comment by spessbro** [comment]
i didn't add that either

**Comment by ATH1909** [comment]
> and rotatium lol

It's SPEWium, not SKEWium in those garlics.



> Gaseous decomp removal would be necessary if he's adding the plasma garlic, otherwise plasma smoke grenade plants would be a thing.
> 
> Spewium is a traitor chem that can make you eject your heart, so i'd be careful with that

Plasma smoke grenade plants won't be a thing if he moves plasma from the contents of the plant into the juice list.

Oh, and spessbro, don't forget to add nutriment production to the plasma onion if you haven't already when you move plasma into the juice list.

**Comment by PKPenguin321** [comment]
The spewium one is a horrible idea

**Comment by granpawalton** [comment]
Some of these are cool like the lyme and treant shell but alot of them look like stuff you would find on hippie

**Comment by tgstationOfficial1SecondFlashbangHater** [comment]
botany is getting dangerously close to xenobio levels of nonsense

**Comment by Ty-the-Smonk** [comment]
Botany is just green xenobio

**Comment by FantasticFwoosh** [comment]
Most of these seem uninspired, I would stay for the lye-me, but you could throw the pod seed mutations & garlic mutations (both of each) into loot/xenobio only or off the list entirely and nobody would bat a eyelid.

**Comment by ExcessiveUseOfCobblestone** [comment]
> i didn't add that either

image

die

**Comment by bgobandit** [comment]
I like the idea of new mutations but a lot of these seem random and/or nonsensical or only exisiting for the sake of a pun.

**Comment by anconfuzedrock** [comment]
if anything it'd be better if replica pods could only be gained as a mutation, removing the cabbage mutation is a stupid idea especially if it's solely so the ability to make itching powder cabbage (even more redundant and out of place) is easier.

**Comment by FantasticFwoosh** [comment]
> if anything it'd be better if replica pods could only be gained as a mutation :: especially if it's solely so the ability to make itching powder cabbage.

Already tried that, community was too resistant. #29933

**Comment by ExcessiveUseOfCobblestone** [comment]
probably would have gotten better results if you weren't fwoosh

**Comment by Tlaltecuhtli** [comment]
why not plants that shoot u

**Comment by ATH1909** [comment]
> if anything it'd be better if replica pods could only be gained as a mutation, removing the cabbage mutation is a stupid idea especially if it's solely so the ability to make itching powder cabbage (even more redundant and out of place) is easier.

God no. Replica pods are already underused as they are; doing that would make them even more obscure.

**Comment by anconfuzedrock** [comment]
okay, let's not make replica pods more obscure. got it. then don't remove the ability to mutate it from cabbage for a stupid fucking pun to be more common.

**Comment by habader** [comment]
stop removing things

File: code/modules/hydroponics/grown/mushrooms.dm [hunk 3]
```diff
@@ -199,6 +231,31 @@
...
+	lifespan = 35
+	endurance = 20
+	maturation = 10
+	production = 12
+	yield = 1
+	potency = 5
+	growthstages = 2
+	genes = list(/datum/plant_gene/trait/plant_type/fungal_metabolism)
+	growing_icon = 'icons/obj/hydroponics/growing_mushrooms.dmi'
+	reagents_add = list(/datum/reagent/consumable/nutriment = 0.1,/datum/reagent/consumable/pricy=0.05)
```
Comment (granpawalton) [inline code/modules/hydroponics/grown/mushrooms.dm]: I think you meant to set the pricy reagent to the truffle_oil reagent, also you should probably make truffle_oil in food_reagents not drink_reagents

**Comment by granpawalton** [comment]
@anconfuzedrock I dont think people were trying to argue for the crab thing I think they were just wanting to keep podpeople as a round start due to its low use

Crabs arent plants so why should we be able to grow them though

Also with the pineapple pizza plant, cooked food should be chefs job for making and I dont think that should be skipped for a meme

**Comment by optimumtact** [comment]
holy shit it's like I have seven goofs on the codebase now

**Comment by ExcessiveUseOfCobblestone** [comment]
> holy shit it's like I have seven goofs on the codebase now
> 

goof is hydra, you kill 1 you get 2 more

File: code/modules/hydroponics/grown/mushrooms.dm [hunk 4]
```diff
@@ -109,6 +110,37 @@
...
+
+/obj/item/reagent_containers/food/snacks/grown/mushroom/brigcap
+	seed = /obj/item/seeds/liberty/brig
+	name = "brig-cap"
+	desc = "<B>Like it or not you're comming with me!</B>"
+	icon_state = "brigcap"
+	filling_color = "#DAA520"
+	wine_power = 80
+	distill_reagent = /datum/reagent/consumable/ethanol/beepsky_smash
+/obj/item/reagent_containers/food/snacks/grown/mushroom/brigcap/attackby(obj/item/W, mob/user, params)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/hydroponics/grown/mushrooms.dm]: this needs inher

File: code/modules/hydroponics/grown/mushrooms.dm [hunk 5]
```diff
@@ -109,6 +110,37 @@
...
+	rarity = 30
+
+/obj/item/reagent_containers/food/snacks/grown/mushroom/brigcap
+	seed = /obj/item/seeds/liberty/brig
+	name = "brig-cap"
+	desc = "<B>Like it or not you're comming with me!</B>"
+	icon_state = "brigcap"
+	filling_color = "#DAA520"
+	wine_power = 80
+	distill_reagent = /datum/reagent/consumable/ethanol/beepsky_smash
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/hydroponics/grown/mushrooms.dm]:
```suggestion
	distill_reagent = /datum/reagent/consumable/ethanol/beepsky_smash
	
```

Comment (ExcessiveUseOfCobblestone) [reply code/modules/hydroponics/grown/mushrooms.dm]: sep 138/139

File: code/modules/hydroponics/grown/mushrooms.dm (hunk 2 above)
Comment (ExcessiveUseOfCobblestone) [inline code/modules/hydroponics/grown/mushrooms.dm]: FYI the variable is `breakouttime`, you could have these scale with potency (100 pot = same as normal handcuffs).

File: code/modules/hydroponics/grown/mushrooms.dm (hunk 3 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/hydroponics/grown/mushrooms.dm]: what is a "pricy"?

File: code/modules/hydroponics/grown/replicapod.dm [hunk 6]
```diff
@@ -132,3 +133,238 @@
...
+	species = "treantshell"
+	plantname = "Treant Shell"
+	product = /mob/living/carbon/human/species/golem/wood
+	rarity = 30
+/obj/item/seeds/replicapod/treantshell/Initialize()
+	. = ..()
+
+	create_reagents(volume, INJECTABLE|DRAWABLE)
+
+/obj/item/seeds/replicapod/treantshell/on_reagent_change(changetype)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/hydroponics/grown/replicapod.dm]: why is this being overwritten?

File: code/modules/hydroponics/grown/replicapod.dm [hunk 7]
```diff
@@ -132,3 +133,238 @@
...
+		mind = null
+		ckey = null
+		realName = null
+		blood_gender = null
+		blood_type = null
+		features = null
+		factions = null
+		contains_sample = FALSE
+
+/obj/item/seeds/replicapod/treantshell/get_analyzer_text()
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/hydroponics/grown/replicapod.dm]: same here

File: code/modules/hydroponics/grown/replicapod.dm [hunk 8]
```diff
@@ -132,3 +133,238 @@
...
+	var/text = ..()
+	if(contains_sample)
+		text += "\n It contains a blood sample!"
+	return text
+
+
+/obj/item/seeds/replicapod/treantshell/harvest(mob/user) //me love copy-paste
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/hydroponics/grown/replicapod.dm]: same here

File: code/modules/hydroponics/grown/replicapod.dm [hunk 9]
```diff
@@ -132,3 +133,238 @@
...
+	product = /mob/living/carbon/human/species/mush
+	rarity = 30
+	mutatelist = list(/obj/item/seeds/replicapod)
+	genes = list(/datum/plant_gene/trait/plant_type/fungal_metabolism)
+/obj/item/seeds/replicapod/mushroompod/Initialize()
+	. = ..()
+
+	create_reagents(volume, INJECTABLE|DRAWABLE)
+
+/obj/item/seeds/replicapod/mushroompod/on_reagent_change(changetype)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/hydroponics/grown/replicapod.dm]: same here...

File: code/modules/hydroponics/grown/replicapod.dm [hunk 10]
```diff
@@ -132,3 +133,238 @@
...
+		mind = null
+		ckey = null
+		realName = null
+		blood_gender = null
+		blood_type = null
+		features = null
+		factions = null
+		contains_sample = FALSE
+
+/obj/item/seeds/replicapod/mushroompod/get_analyzer_text()
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/hydroponics/grown/replicapod.dm]: same here...

File: code/modules/hydroponics/grown/replicapod.dm [hunk 11]
```diff
@@ -132,3 +133,238 @@
...
+	var/text = ..()
+	if(contains_sample)
+		text += "\n It contains a blood sample!"
+	return text
+
+
+/obj/item/seeds/replicapod/mushroompod/harvest(mob/user) //me love copy-paste
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/hydroponics/grown/replicapod.dm]: same here...

File: code/modules/reagents/chemistry/reagents/food_reagents.dm [hunk 12]
```diff
@@ -729,4 +729,13 @@
 	taste_description = "indescribable"
 	quality = FOOD_AMAZING
 	taste_mult = 100
+	can_synth = FALSE
+
+/datum/reagent/consumable/pricy
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/reagents/food_reagents.dm]: what lol

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]

**Comment by spessbro** [comment]
i'm getting a lot of irl troubles so imma go finish this another day sorry

**Pull request closed without merging**